[CS2113-T11-3] CliRental

tkhahns opened the pull request:
CliRental helps manage a car rental business that handles large volumes of data, making traditional pen-and-paper methods inefficient for managers. This app is designed specifically for car rental managers, offering a Command Line Interface (CLI) that enables quick data entry and efficient tracking of customers, cars, and transactions.

tzerbin commented on src/main/java/customer/CustomerList.java:
```diff
@@ -0,0 +1,36 @@
+
+public class CustomerList {
+
+    public static ArrayList<Customer> customers = new ArrayList<>();
```
Inconsistent naming through the codebase: customers v.s. carsList.
It would be good if this is consistent throughout the codebase.

tzerbin commented on src/main/java/parser/CustomerParser.java:
```diff
@@ -0,0 +1,86 @@
+        String username = parameterContents[0];
+        int age = Integer.parseInt(parameterContents[1]);
+        int contactNumber = Integer.parseInt(parameterContents[2]);
+        return new Customer(username , age, contactNumber );
```
Be mindful about the horizontal whitespace you use throughout the codebase.
```suggestion
        return new Customer (username , age, contactNumber);
```

tzerbin commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,100 @@
+        return userInput;
+    }
+
+    public static boolean parse(String userInput) throws CliRentalException {
```
Boolean methods should start with is/has/etc.
Consider using more exceptions and asserts to assist with making sure the method runs successfully.

tzerbin commented on src/main/java/seedu/clirental/CliRental.java:
```diff
@@ -0,0 +1,54 @@
+        System.out.println("Goodbye!");
+    }
+
+    public static void getName(){
```
Readers would normally expect a Name return type for a method named getName.
Consider renaming this method.
This issue is apparent throughout the codebase.

tzerbin commented on src/main/java/transcation/Transaction.java:
```diff
@@ -0,0 +1,47 @@
+
+public class Transaction {
+    private static int transactionCounter = 1;
+    private final String transactionId;
```
Final attributes should be named with SCREAMING_SNAKE_CASE.

tzerbin commented on src/main/java/transcation/TransactionList.java:
```diff
@@ -0,0 +1,57 @@
+        return false;
+    }
+
+    public static void addTx(Transaction transaction) {
```
Be consistent with the naming "Tx" and "transactions". The latter would be more appropriate.

tzerbin commented on src/test/java/customer/CustomerListTest.java:
```diff
@@ -0,0 +1,22 @@
+public class CustomerListTest {
+
+    @Test
+    public void testAddCustomer() {
```
Please follow the naming convention for tests.

tzerbin reviewed (commented):
Good effort in following the coding standards!
Please make the necessary changes such that it follows more closely!

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,43 @@
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links 
+to the original source as well}
```
Could add a contents page here for easy reference for the DG

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,43 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The following is our class diagram for our whole project.
 
+![Local Image](UML/CLiRental.png)
```
Should you add all the other classes in your main/java/car folder?

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,43 @@
+
+The following sequence diagram will explain the sequence of events after the user inputs an add-user command.
+
+![Local Image](UML/AddCustomerSequence.png)
```
Nice use of numbers to explain sequence of events for long and complicated diagrams

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,43 @@
+company/store. It also allows for the staff of the rental company to filter through the massive amount of transactions,
+finding the transaction they are looking for easily with multiple filters.
 
 ## User Stories
```
Could update these sections for the final version as they are needed

coraleaf0602 commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,43 @@
+loaded if the file exist.
+
+![Local Image](UML/CarFileLoader.png)
 
```
I would consider adding a few more UML diagrams if possible. 

follow the format of the AB3 
https://se-education.org/addressbook-level3/UserGuide.html

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The following is our class diagram for our whole project.
```
An architecture diagram would be more suitable to describe the whole project.
This class diagram can be used to discuss how Customer class is being used in the project.

CliRental class seems to be very out of place here.
Do take note of the usage "Void" v.s. "void", "String" v.s. "string".
Do include multiplicity as well.
The class diagram seems to suggest aggregation instead of composition.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+The following sequence diagram will explain the sequence of events after the user inputs an add-user command.
+
+![Local Image](UML/AddCustomerSequence.png)
```
Good attempt on the sequence diagrams.
However, there may be too many information within the diagram. Perhaps omit details that involve the Java API e.g. method calls to ArrayList.

Ridiculouswifi replied:
![image](https://github.com/user-attachments/assets/6c9c496e-e4b8-4dd1-a8e3-607f95d309bd)

Perhaps line 14 should be dotted instead? Given that it shows the function CustomerException returning to the activation bar of CustomerParser.

![image](https://github.com/user-attachments/assets/dc50273f-7f3a-4703-b8a5-344be00ca918)

Seems like the arrow for line 21 is in the wrong direction.

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+The following sequence diagram illustrates a **valid** `add-car` operation:
+
+![](./images/RyanAddCarDiagram.png)
```
Is there a reason why there are different software for creating sequence diagrams?

tzerbin commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+The following class diagram shows the interaction between the classes involved:
+
+![](./images/RyanClassDiagram.png)
```
It would be better to explain the classes involved first before showing sequence diagrams.
Multiplicities can be improved to both sides of an association.
Double-check with the composition: If CarList is destroyed, why would instances of Cars be destroyed too?
Double-check with dependencies: Does CarParser use CarList?

tzerbin reviewed (commented):
Good attempt on updating your DG.
There are small issues pertaining your diagrams, please make sure that the diagrams communicate how your software is implemented.
Some information are missing due to lack of architecture / class diagrams.
Do complete your DG - some sections are empty.

Ridiculouswifi commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+![](./images/list-rented-output.png)
+
+Step 5: **Optionally**, the user can also execute 
```
I like the explanation of this process in steps, it helps to make it more readable.

Ridiculouswifi commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+
+Step 3: A customer decides to rent a car from the company. The user then uses the application to track
+and record the transaction details. The user executes the `add-tx ...` command to add
```
Perhaps it would be better to have an example for this command, as it is not mentioned before in this document.

Ridiculouswifi commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+Example:
+
+![](./images/list-cars-output-before.png)
```
Perhaps it would be better to use the same software for creating sequence diagrams to ensure consistency.

Ridiculouswifi reviewed (commented):
Good first draft! When updating the developer guide, I think it would be good to adhere to the same format if possible.

Ridiculouswifi commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The following is our class diagram for our whole project.
 
+![Local Image](UML/CLiRental.png)
```
Perhaps it would be better to leave out the minor utility methods such as the getters and setters as the class diagram does look quite cluttered.

Ridiculouswifi commented on docs/DeveloperGuide.md:
```diff
@@ -2,21 +2,133 @@
+
+![Local Image](UML/AddCustomerSequence.png)
+
+2. `Creating Car file and loading file`
```
![image](https://github.com/user-attachments/assets/3f3b3f9b-2ab7-4a93-b074-a1af99185227)

Seems like there was an error when rendering the document, it should be 2. instead of 1.

rexkoh425 committed:
Fix issue #138

rexkoh425 committed:
Fix issue #142

tkhahns committed:
Update Test finding transactions by customer

rexkoh425 committed:
Fix issue #153

tkhahns committed:
Fix test case

tkhahns committed:
Modify test case logic

rexkoh425 committed:
Add manual testing instructions

Commit:
Update files

Commit:
Update files

Commit:
solve tx id not generated when addtxwithoutPrintingInfo is called

tkhahns committed:
Update TransactionListTest.java

tkhahns committed:
Update TransactionListTest.java

rexkoh425 committed:
Add age restriction for customer

rexkoh425 committed:
Update Junit

Commit:
added generation of transaction id to test cases

Commit:
extracted if conditions for checks on car has been rented during and customer has a rental for a new add-tx rental

rexkoh425 committed:
Update Junit

Update adding of transaction , car , customer without printing info

CT9ARyan committed:
Add new exception to handle cases when car rental price exceeds the threshold limit

CT9ARyan committed:
Define new constant to store maximum car rental price

CT9ARyan committed:
Update 'isValidPrice' method and edit 'isValidPrice' code block in 'parseIntoCar' to throw the appropriate exception

rexkoh425 committed:
Update Quick Start

rexkoh425 committed:
Update imports

SemiColonKen committed:
Merge pull request #207 from SemiColonKen/master

Improve A-AddRental and Transaction_ID generation

rexkoh425 committed:
Merge branch 'master' into A-SaveFileDocumentation

rexkoh425 committed:
Fix checkstyle

rexkoh425 committed:
Merge branch 'A-SaveFileDocumentation' of https://github.com/rexkoh425/tp into A-SaveFileDocumentation

rexkoh425 committed:
Fix checkstyle

rexkoh425 committed:
Update Ui test

rexkoh425 committed:
Update Ui test

rexkoh425 committed:
Update Ui test

rexkoh425 committed:
Change to Customer name

rexkoh425 committed:
Merge pull request #211 from rexkoh425/A-SaveFileDocumentation

A-SaveFileDocumentation

AaronZZ10 committed:
fix authorship

AaronZZ10 committed:
fix authorship

tkhahns committed:
Merge branch 'master' into master

AaronZZ10 committed:
updated ppp

AaronZZ10 committed:
Merge pull request #212 from AaronZZ10/master

Update PPP

tkhahns committed:
update clearTransactionList mehod

rexkoh425 committed:
Update to store transactionID

tkhahns committed:
Fix testLoadTransactionDataIfExist()

tkhahns committed:
Fix testUpdateTransactionDataFile()

rexkoh425 committed:
Remove unused imports

tkhahns committed:
Fix testGetTransactionDataFilename()

tkhahns committed:
Merge pull request #213 from rexkoh425/A-StoreTxID

A-StoreTxId

tkhahns committed:
Merge branch 'master' into master

tkhahns committed:
Fix Transactions code

Commit:
Change implementation of Tx ID generation for easier saving and loading. Fixed Tx Parser to check for spaces.

Commit:
Add Transactionfile to load Tx Counter

tkhahns committed:
Merge pull request #208 from tkhahns/master

Update transactions constraints

rexkoh425 committed:
Add error handling for data

rexkoh425 committed:
Fix checkstyle

rexkoh425 committed:
Update UG

Commit:
Fixed saved tx id

SemiColonKen committed:
Merge branch 'master' into master

Commit:
Fixed junit test error

Commit:
Fixed junit test error (forgot to add transaction files)

Commit:
Fixed merge error (TransactionList)

Commit:
Fixed junit test error (TransactionTest)

Commit:
Fixed junit test error (TransactionListTest)

Commit:
Fixed junit test error (TransactionTest)

tkhahns committed:
Merge pull request #215 from SemiColonKen/master

Fixed Tx ID saving and loading

tkhahns committed:
Change phone number format

tkhahns committed:
Merge branch 'master' of https://github.com/tkhahns/tp

tkhahns committed:
A-RemoveAllCustomers

tkhahns committed:
Remove clearTransactionCounter method in test file

tkhahns committed:
Add phone number constraints

Singapore phone numbers only (8 digits, starts with either 8 or 9)

tkhahns committed:
Add test cases for valid phone numbers

tkhahns committed:
Fix parser test

tkhahns committed:
Comment out the clear method before tests

tkhahns committed:
Remove clear methods

tkhahns committed:
Update EXPECTED output

tkhahns committed:
Modify runtest.sh

tkhahns committed:
Fix runtest.bat and find-txs-by-customer

CT9ARyan committed:
Update documentation and sample code output for some features in user guide

Commit:
Fixed bug: find-tx-by-customer to display exact input from customer and not lowercase input

CT9ARyan committed:
Restore background color of all diagrams to white to standardise across all diagrams in the developer guide

tkhahns committed:
Modify EXPECTED output

CT9ARyan committed:
Update 'CarList' methods to ignore letter case when querying for license plate number in the list

CT9ARyan committed:
Add new method to check if string can be parsed to double in 'CarParser'

Commit:
Fixed junit test error (TransactionListTest)

CT9ARyan committed:
Add new if block that throws NumberFormatException with a custom message if the price string cannot be parsed to a double type

SemiColonKen committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

tkhahns committed:
Merge pull request #216 from tkhahns/master

A-RemoveAllCustomers

CT9ARyan committed:
Edit 'remove-car' methods and related exception messages when an exception is thrown

CT9ARyan committed:
Edit commands description in help page to match those in user guide

CT9ARyan committed:
Add try-catch block for the add-car command case to catch any NumberFormatException thrown from CarParser

CT9ARyan committed:
Update command format of add-tx

Command format parameter example changed from CAR_ID to LICENSE_PLATE_NUMBER to standardise across the app and the user guide.

CT9ARyan committed:
Update 'removeAllTxs' method to update all car rental status to 'Available' before clearing transactions

Commit:
Fixed bug; Duration not shown when add-tx with 1 day

CT9ARyan committed:
Merge branch 'master' into branch-BugFixes

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/parser/HelpParser.java
#	src/main/java/parser/TransactionParser.java

tkhahns committed:
Merge pull request #217 from SemiColonKen/master

Fixed bug: find-tx-by-customer & duration not shown when add-tx with 1 day

CT9ARyan committed:
Add summary of contents with section links in user guide

CT9ARyan committed:
Update JUnit tests and test conditions

CT9ARyan committed:
Update EXPECTED.txt

CT9ARyan committed:
Add input.txt back to text-ui-test folder due to accidental deletion

CT9ARyan committed:
Edit input.txt contents

CT9ARyan committed:
Update EXPECTED.txt

SemiColonKen committed:
Merge pull request #218 from CT9ARyan/branch-BugFixes

A-BugFixesV1

tkhahns committed:
Update unique Customer constraints

tkhahns committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

CT9ARyan committed:
Update 'add-car' description to be clearer and more specific in user guide

CT9ARyan committed:
Update portfolio link to PPP in AboutUs.md

CT9ARyan committed:
Add new .md file for PPP

CT9ARyan committed:
Add 'Overview' section documenting product summary in PPP

CT9ARyan committed:
Add 'Code contribution' and 'Features implemented' sections in PPP

CT9ARyan committed:
Add 'User guide contributions' in PPP

CT9ARyan committed:
Add 'Developer guide contributions' section in PPP

tkhahns committed:
Fix checkstyle error

tkhahns committed:
Merge branch 'master' of https://github.com/tkhahns/tp

tkhahns committed:
Fix checkstyle error

tkhahns committed:
Merge pull request #219 from tkhahns/master

CT9ARyan committed:
Add product name at the top of PPP

CT9ARyan committed:
Add pull request links for PRs reviewed under 'Team-based tasks contributions' section in PPP

CT9ARyan committed:
Improve format and aesthetics of PPP and edited description of some features

Commit:
Reset TX ID counter to 1 when removing all TXs

rexkoh425 committed:
Merge branch 'master' into A-RexCodeQuality

rexkoh425 committed:
Resolve merge conflict

tkhahns committed:
Add test cases

rexkoh425 committed:
Hide footnote

rexkoh425 committed:
Edit Junit

tkhahns committed:
Merge pull request #222 from tkhahns/master

Add test cases

tkhahns committed:
Merge pull request #220 from SemiColonKen/master

Reset TX ID counter to 1 when removing all TXs

rexkoh425 committed:
Resolve merge conflict

rexkoh425 committed:
Resolve merge conflict

CT9ARyan committed:
Convert newly updated UML diagrams to .png file to be used in developer guide

Background color changed to white to standardize with other diagrams in developer guide.

Commit:
Updated UG for missing commands. Added updated sample output.

rexkoh425 committed:
Update contact number info

tkhahns committed:
Merge pull request #223 from SemiColonKen/master

Updated UG for missing commands. Added updated sample output.

Commit:
Updated UG

rexkoh425 committed:
Fix spacing

rexkoh425 committed:
Merge branch 'master' into A-RexCodeQuality

rexkoh425 committed:
Merge pull request #221 from rexkoh425/A-RexCodeQuality

A-RexCodeQuality

tkhahns committed:
Merge pull request #224 from SemiColonKen/master

Updated UG

tkhahns committed:
Create PPP

CT9ARyan committed:
Update list of commands in help page

CT9ARyan committed:
Add new UML sequence diagram for 'remove-car' command

CT9ARyan committed:
Remove box grouping in remove car sequence diagram

CT9ARyan committed:
Convert remove car sequence diagram into .png for developer guide

rexkoh425 committed:
Update diagrams

CT9ARyan committed:
Update code output .png files due to change in code output format for DG

CT9ARyan committed:
Add new section for 'removing a car' sequence diagram

CT9ARyan committed:
Remove unnecessary lines in DG

Commit:
add-tx customer name is case insensitive now. John and john means the same thing. The actual name from the customer list will be used in add-tx for neatness.

SemiColonKen committed:
Merge branch 'master' into master

CT9ARyan committed:
Merge branch 'master' into branch-PPP

# Conflicts:
#	docs/DeveloperGuide.md

rexkoh425 committed:
Update DG

rexkoh425 committed:
Edit checkstyle

CT9ARyan committed:
Update EXPECTED.txt to pass CLI test on github

SemiColonKen committed:
Merge pull request #226 from SemiColonKen/master

Add-TX improvement which ignores the case of Customer Name

CT9ARyan committed:
Merge pull request #225 from CT9ARyan/branch-PPP

A-DocumentationAndBugFixesUpdate

tkhahns committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

rexkoh425 committed:
Update PPP

rexkoh425 committed:
Update sequence diagram

rexkoh425 committed:
Add profile pic

Commit:
remove-user is now case insensitive

Commit:
remove-user is now case insensitive

SemiColonKen committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

tkhahns committed:
Slight modification of output for printUncompletedTransactions, removeCustomer

tkhahns committed:
Fix userName not found test case

tkhahns committed:
Merge pull request #228 from tkhahns/master

Fix very low severity bug that I found myself

rexkoh425 committed:
Improve Junit test case

SemiColonKen committed:
Merge branch 'master' into master

rexkoh425 committed:
Merge pull request #229 from rexkoh425/A-Rex-V2-1

A-RexV2.1

SemiColonKen committed:
Merge pull request #227 from SemiColonKen/master

remove-user is case insensitive now

CT9ARyan committed:
Update formatting for PPP

tkhahns committed:
Update logo and fix find-txs-by-customer output

tkhahns committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

tkhahns committed:
Fix checkstyle error

tkhahns committed:
Update EXPECTED.txt

tkhahns committed:
Update logo

tkhahns committed:
Complete DG with User Stories

tkhahns committed:
Update AboutUs.md

tkhahns committed:
Add PPP

tkhahns committed:
Update PPP

Commit:
updated ug

Commit:
updated customer list

Commit:
Merge branch 'master' of https://github.com/SemiColonKen/tp

Commit:
Added PPP

Commit:
Updated remove-customer

Commit:
add junittest for remove-user

Commit:
updated remove-user for empty list

AaronZZ10 committed:
updated ppp

AaronZZ10 committed:
improve code quality

Commit:
updated import for CustomerListTest to fix checkstyle error

Commit:
updated line length for CustomerList to fix checkstyle error

AaronZZ10 committed:
improve code quality

AaronZZ10 committed:
improve code quality

AaronZZ10 committed:
Merge pull request #232 from AaronZZ10/master

Improve Code Quality

tkhahns committed:
Update PPP

tkhahns committed:
Merge pull request #230 from tkhahns/master

Update logo and fix find-txs-by-customer output

tkhahns committed:
Merge pull request #231 from SemiColonKen/master

Update UG to be more detailed. Added junittest for remove-user. Updated remove-user for empty customer list. Added PPP

rexkoh425 committed:
Update documentation

tkhahns committed:
Merge pull request #234 from rexkoh425/A-RexFinalPR

A-UpdateDocumentation

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
update iotest

rexkoh425 committed:
Update Junit

AaronZZ10 committed:
update iotest

rexkoh425 committed:
Merge pull request #236 from rexkoh425/A-JunitFix

Update Junit

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
update iotest

rexkoh425 committed:
Update Junit

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
update iotest

rexkoh425 committed:
Update Junit

AaronZZ10 committed:
update iotest

rexkoh425 committed:
Update DG for manual test

AaronZZ10 committed:
update iotest

AaronZZ10 committed:
update iotest

tkhahns committed:
Merge pull request #237 from rexkoh425/A-JunitFix

Update Junit

AaronZZ10 committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

rexkoh425 committed:
Update age limit handling

AaronZZ10 committed:
Merge pull request #235 from AaronZZ10/master

update iotest

tkhahns committed:
Merge pull request #238 from rexkoh425/A-JunitFix

A-JunitFixAge

AaronZZ10 committed:
fix merge conflict

rexkoh425 committed:
Update PPP

rexkoh425 committed:
Merge pull request #239 from rexkoh425/A-JunitFix

A-UpdateRexPPP

rexkoh425 committed:
Add <br>

rexkoh425 committed:
Merge pull request #240 from rexkoh425/A-JunitFix

Update UG

rexkoh425 committed:
Update formatting

AaronZZ10 committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

AaronZZ10 committed:
images of ryan and aaron

rexkoh425 committed:
Merge pull request #242 from rexkoh425/A-JunitFix

Update formatting

AaronZZ10 committed:
Merge pull request #241 from AaronZZ10/master

images of ryan and aaron

Commit:
Updated ug and ppp

tkhahns committed:
Slight modification of UG, DG, and upload team members' photos

tkhahns committed:
Merge pull request #243 from SemiColonKen/master

Updated ug and ppp

tkhahns committed:
Merge branch 'master' into master

tkhahns committed:
Merge pull request #244 from tkhahns/master

Slight modification of UG, DG, and upload team members' photos

AaronZZ10 committed:
update ppp and dg

AaronZZ10 committed:
Merge pull request #245 from AaronZZ10/master

update ppp and dg

AaronZZ10 committed:
update ppp and dg

AaronZZ10 committed:
update ppp

AaronZZ10 committed:
update ppp

AaronZZ10 committed:
update ppp

AaronZZ10 committed:
update ppp

AaronZZ10 committed:
Merge pull request #246 from AaronZZ10/master

update ppp

tkhahns committed:
Update DG

tkhahns committed:
Merge branch 'AY2425S1-CS2113-T11-3:master' into master

AaronZZ10 committed:
update ppp

tkhahns committed:
Merge pull request #247 from tkhahns/master

DG

AaronZZ10 committed:
Merge pull request #248 from AaronZZ10/master

update ppp

tkhahns committed:
Update AboutUs.md

tkhahns committed:
Merge branch 'master' of https://github.com/tkhahns/tp

tkhahns committed:
Merge pull request #249 from tkhahns/master

Update AboutUs.md - Final PR

tkhahns committed:
Fix Kenneth's PPP link error

tkhahns committed:
Merge pull request #250 from tkhahns/master

Fix Kenneth's PPP link error
